src/Database/DB.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Bonnier\WP\Redirect\Database;
+
+use Bonnier\WP\Redirect\Database\Exceptions\DuplicateEntryException;
+use Illuminate\Support\Str;
+
+class DB
+{
+    /** @var \wpdb */
+    private $wpdb;
+    /** @var string */
+    private $table;
+
+    public function __construct()
+    {
+        global $wpdb;
+        $this->wpdb = $wpdb;
+        $this->wpdb->show_errors(false);
+        $this->wpdb->suppress_errors(true);
+    }
+
+    /**
+     * @param string $tableName
+     * @return string
+     */
+    public function setTable(string $tableName)
+    {
+        return $this->table = Str::start($tableName, $this->wpdb->prefix);
+    }
+
+    /**
+     * @param int $rowID
+     * @return array|null
+     */
+    public function findById(int $rowID): ?array
+    {
+        return $this->wpdb->get_row(
+            $this->wpdb->prepare("SELECT * FROM $this->table WHERE id = %d", $rowID),
+            ARRAY_A
+        );
+    }
+
+    /**
+     * @return Query
+     */
+    public function query(): Query
+    {
+        return new Query($this->table);
+    }
+
+    /**
+     * @param Query $query
+     * @return array
+     * @throws \Exception
+     */
+    public function getResults(Query $query): array
+    {
+        return $this->wpdb->get_results($query->getQuery(), ARRAY_A);
```

Comment:
Perhaps rename **getQuery**? Like what does getQuery return? a string or an array or?
Maybe **toSq** if it returns a string? 
```suggestion
        return $this->wpdb->get_results($query->toSql(), ARRAY_A);
```